### PR TITLE
reformatting optimizer update and fix stop_gradient out attribute error

### LINKED ISSUE #1833: optimizer_update

### PR DESCRIPTION
**Pull request by yugaljain1999**
fixes #1833 and stop_gradient has no attribute "out" error.

### CODE REVIEW

**Review by djl11 — CHANGES_REQUESTED** [review]
all examples need to be aligned, have set framework removed, and there should be less examples overall. Please check out at the deep dive section in the docs.

File: ivy/functional/ivy/gradients.py [hunk 1]
```diff
@@ -400,6 +400,110 @@ def optimizer_update(
...
+    >>> print(wsnew)
+    ivy.array([[0.999, 1.95 ],
+            [4.   , 4.92 ]])
+    
+    >>> ivy.set_backend('torch')
+    >>> w = ivy.array([1., 2., 3.])
+    >>> effective_grad = ivy.zeros(3)
+    >>> lr = 3e-4
+    >>> wsnew = ivy.optimizer_update(w=w, effective_grad=effective_grad, lr=lr,\
+                                    stop_gradients=False)
```
[the remark this answers is not in the document]
Comment (yugaljain1999) [reply ivy/functional/ivy/gradients.py:430]: @djl11 Thanks for your suggestion. May I know what is misaligned here? I couldn't find what's wrong in this.

**Comment by yugaljain1999** [comment]
@djl11  I updated docstrings examples as per you change request. Lemme know if it's good to go now.
Thanks

**Review by djl11 — CHANGES_REQUESTED** [review]
remove .pyc files, and you are not using all arguments in the functional examples, for example the inplace argument. Check the deep dive docstring examples section again.

**Comment by yugaljain1999** [comment]
@djl11 I just updated docstrings examples and removed .pyc as you requested.
Can you pls review this PR now?
Thanks

File: ivy/functional/ivy/gradients.py [hunk 2]
```diff
@@ -37,17 +37,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 # Gradient Mode #
 
 # noinspection PyShadowingNames
-def with_grads(with_grads=None):
-    """Summary.
+def with_grads(with_grads: bool = None) -> bool:
+    """
+    Enter a nested code space where gradients are computed. This method
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:42]: what is "nested" exactly?

File: ivy/functional/ivy/gradients.py [hunk 3]
```diff
@@ -37,17 +37,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+    adds the with_grads component to the global list with_grads_stack
 
     Parameters
     ----------
     with_grads
-         (Default value = None)
+        Boolean value denoting whether the current code block has gradient
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:48]: where is the code block? If it corresponds to code blocks, why are there not code blocks in the example?

File: ivy/functional/ivy/gradients.py [hunk 4]
```diff
@@ -37,17 +37,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+        computation enabled or not.
+        'True' or 'False' or 'None' (Default value = None)
 
     Returns
     -------
     ret
+        If with_grads is boolean, it returns the boolean value representing
+        if gradient computation is enabled or not.
+        If with_grads is None, it returns the last element in the with_grads_stack
+        representing the parent of the current nested code block. If with_grads_stack
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:58]: again, let's show nested code blocks in the examples if this is what the function is for.

File: ivy/functional/ivy/gradients.py [hunk 5]
```diff
@@ -400,6 +426,63 @@ def optimizer_update(
...
         The new function weights ws_new, following the optimizer updates.
+    
+    Functional Examples
+    -------------------
+    With :code:`ivy.Array` inputs:
+    
+    >>> w = ivy.array([1., 2., 3.])
+    >>> effective_grad = ivy.zeros(3)
+    >>> lr = 3e-4
+    >>> wsnew = ivy.optimizer_update(w=w, effective_grad=effective_grad, lr=lr,\
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:437]: use `ws_new`, we should have underscores between words in a variable name.

File: ivy/functional/ivy/gradients.py [hunk 6]
```diff
@@ -400,6 +426,63 @@ def optimizer_update(
     -------
     ret
         The new function weights ws_new, following the optimizer updates.
+    
+    Functional Examples
+    -------------------
```
Comment (djl11) [inline ivy/functional/ivy/gradients.py:431]: we don't show any examples where `inplace` and `stop_gradients` are unspecified. We should include examples where the default values are implicitly used.

**Review by djl11 — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/gradients.py (hunk 4 above)
Comment (yugaljain1999) [reply ivy/functional/ivy/gradients.py:58]: @djl11 I didn't change this, I just pulled latest code from master branch and then just added docstrings examples of optimizer_update function for which I made PR.

File: ivy/functional/ivy/gradients.py (hunk 6 above)
Comment (yugaljain1999) [reply ivy/functional/ivy/gradients.py:431]: Yeah I exactly added those examples only where default values are implicitly used.
You can refer docstring examples from line 429 onwards for optimizer_update function.

File: ivy/functional/ivy/gradients.py (hunk 5 above)
Comment (yugaljain1999) [reply ivy/functional/ivy/gradients.py:437]: Sure I will commit for this change

File: ivy/functional/ivy/gradients.py (hunk 6 above)
Comment (djl11) [reply ivy/functional/ivy/gradients.py:431]: please re-read the comment above and add examples without these keyword arguments provided.

**Review by djl11 — CHANGES_REQUESTED** [review]

**Comment by yugaljain1999** [comment]
@djl11 I just added some examples without those keyword arguments provided. Can you pls review it now?
Thanks

**Comment by djl11** [comment]
Please create a new PR with only the reformatting performed for a single function. There is too much going on in this PR.

**Pull request closed without merging**